catalog/app/components/Layout/Layout.tsx

```diff
@@ -42,13 +43,16 @@ export interface LayoutProps {
 export function Layout({ bare = false, dark = false, children, pre }: LayoutProps) {
   const { paths } = NamedRoutes.use()
   const isHomepage = useRouteMatch(paths.home)
+  const bucketRoute = useRouteMatch(paths.bucketRoot)
+  const { bucket } = (bucketRoute?.params as { bucket?: string }) || {}
   return (
     <Root dark={dark}>
       {bare ? <NavBar.Container /> : <NavBar.NavBar />}
       {!!pre && pre}
       {!!children && <M.Box p={4}>{children}</M.Box>}
       <M.Box flexGrow={1} />
       {!!isHomepage && isHomepage.isExact && <Footer />}
+      <Bookmarks.Sidebar bucket={bucket} />
```

Comment:
They are dumped in one group "main", but probably should be organized by user-specified groups (these groups can be buckets as well).

This `bucket` prop is required for creating package. Because I need to create a package in some bucket, and it's reasonable to use the bucket which user browsing right now.

Reply:
> This bucket prop is required for creating package

ah ok makes sense